src/main/java/org/jitsi/jigasi/transcription/TranscriptionService.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Jigasi, the JItsi GAteway to SIP.
+ *
+ * Copyright @ 2017 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jigasi.transcription;
+
+import java.util.function.Consumer;
+
+/**
+ * A TranscriptionService provides the ability to send audio either
+ * as complete audio fragments or as a continuous stream of audio packets
+ * to a speech-to-text API for transcription.
+ *
+ * Fragments of audio are assumed to lie in the range of 15 to 60 seconds
+ * of audio.
+ *
+ * A continuous stream of audio packets has no expected maximum duration.
+ *
+ * @author Nik Vaessen
+ */
+public interface TranscriptionService
+{
+    /**
+     * Get whether this TranscriptionService supports sending fragments of audio
+     *
+     * @return true when fragmented transcription is supported, false otherwise
+     */
+    boolean supportsFragmentTranscription();
+
+    /**
+     * Sends an audio fragment request to the service to be transcribed.
+     * The fragment of audio should contain at least one spoken word for
+     * transcription to be successful. Fragments longer than 60 seconds
+     * might not be supported by most speech-to-text or will require
+     * to much processing time to get a timely answer.
+     *
+     * @param request the TranscriptionRequest which holds the audio
+     * @param resultConsumer a Consumer of the transcription result
+     * @throws UnsupportedOperationException when this service does not support
+     * fragmented audio speech-to-text
+     */
+    void sent(TranscriptionRequest request,
```

Comment:
I will rename it `sendSingleRequest` to emphasise that this method should be used for one audio complete file/fragment